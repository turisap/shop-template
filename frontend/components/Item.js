import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link'
import ItemStyles from './styles/ItemStyles';
import Title from './styles/Title';
import PriceTag from './styles/PriceTag';
import formatMoney from '../lib/formatMoney';
import DeleteItem from './DeleteItem';

class Item extends Component {
    render() {
        const { id, title, price, image } = this.props.item;
        return (
            <ItemStyles>
                {image && <img src={image} alt={title}/>}
                <Title>
                    <Link href={{ pathname : '/item', query : { id } }}>
                        <a>{title}</a>
                    </Link>
                    <PriceTag> {formatMoney(price)} </PriceTag>
                </Title>
                <div className="buttonList">
                    <Link href={{pathname : "update", query : { id }}}>
                        <a>Edit</a>
                    </Link>
                    <button>Add to Cart</button>
                    <DeleteItem id={id}>Delete this item</DeleteItem>
                </div>
            </ItemStyles>
        );
    }
}

Item.propTypes = {
    item : PropTypes.object.isRequired
};

export default Item;
