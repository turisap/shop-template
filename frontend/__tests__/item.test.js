import Item from '../components/Item';
import { shallow } from 'enzyme';

const fakeItem = {
    id : 'laskdjf',
    title : ' A fake item',
    price : 98,
    description : 'Fake fake fake',
    image : 'fake.jpg',
    largeImage : 'fakeLarge.jpg'
};

describe('<Item/>', () => {
    it('renders and displays properly', () => {
        const wrapper = shallow(<Item item={fakeItem}/>);
        const PriceTag = wrapper.find('PriceTag');
        //console.log(wrapper.debug())
        //console.log(PriceTag.debug())
        //console.log(PriceTag.text())
        //console.log('LOLOLOLO ' + PriceTag.children())
        expect(PriceTag.children().text()).toBe('$0.98');
        expect(wrapper.find('Title a').text()).toBe(fakeItem.title);

    });

    it('renders the image properly', () => {
        const wrapper = shallow(<Item item={fakeItem}/>);
        const img = wrapper.find('img');
        //console.log(wrapper.instance().props)
        expect(img.props().src).toBe(fakeItem.image);
        expect(img.props().alt).toBe(fakeItem.title);
    });

    it('renders the button properly', () => {
        const wrapper = shallow(<Item item={fakeItem}/>);
        const buttonList = wrapper.find('.buttonList');
        expect(buttonList.children()).toHaveLength(3);
        expect(buttonList.find('Link').exists()).toBe(true);
        expect(buttonList.find('AddToCart').exists()).toBe(true);
        expect(buttonList.find('DeleteItem').exists()).toBe(true);
    })
})