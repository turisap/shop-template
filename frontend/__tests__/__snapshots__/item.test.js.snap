// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt=" A fake item"
    src="fake.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "laskdjf",
          },
        }
      }
    >
      <a>
         A fake item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $9.86
  </PriceTag>
  <p>
    Fake fake fake
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "laskdjf",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="laskdjf"
    />
    <DeleteItem
      id="laskdjf"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
